code/game/objects/effects/spawners/random/structure.dm

```diff
@@ -0,0 +1,201 @@
+/obj/effect/spawner/random/structure
+	name = "structure spawner"
+	desc = "Now you see me, now you don't..."
+
+/obj/effect/spawner/random/structure/crate
+	name = "crate spawner"
+	icon_state = "crate"
+	loot = list(
+		/obj/effect/spawner/random/structure/crate_empty = 75,
+		/obj/structure/closet/crate/trashcart/filled = 7,
+		/obj/effect/spawner/random/trash/moisture_trap = 5,
+		/obj/effect/spawner/random/trash/hobo_squat = 3,
+		/obj/structure/closet/mini_fridge = 3,
+		/obj/effect/spawner/random/trash/mess = 3,
+		/obj/structure/closet/crate/decorations = 1,
+	)
+
+/obj/effect/spawner/random/structure/crate_abandoned
+	name = "locked crate spawner"
+	icon_state = "crate_secure"
+	spawn_loot_chance = 20
+	loot = list(/obj/structure/closet/crate/secure/loot)
+
+/obj/effect/spawner/random/structure/girder
+	name = "girder spawner"
+	icon_state = "girder"
+	loot = list(
+		/obj/structure/girder = 4,
+		/obj/structure/girder/displaced = 1,
+	)
+
+/obj/effect/spawner/random/structure/grille
+	name = "grille spawner"
+	icon_state = "grille"
+	loot = list(
+		/obj/structure/grille = 4,
+		/obj/structure/grille/broken = 1,
+	)
+
+/obj/effect/spawner/random/structure/lattice
+	name = "lattice spawner"
+	icon_state = "lattice"
+	loot = list(
+		/obj/structure/lattice = 4,
+		/obj/structure/lattice/catwalk = 1,
+	)
+
+/obj/effect/spawner/random/structure/spare_parts
+	name = "spare parts spawner"
+	icon_state = "table_parts"
+	loot = list(
+		/obj/structure/table_frame,
+		/obj/structure/table_frame/wood,
+		/obj/item/rack_parts,
+	)
+
+/obj/effect/spawner/random/structure/table_or_rack
```

Comment:
all this does is spawn another spawner- is something missing from it?

Reply:
Nice catch!  Added rack structure to it with https://github.com/tgstation/tgstation/pull/60522/commits/c28d2ddc5b2669664181a29272ddfae8cc941cf2